### PR TITLE
workaround for require.resolve in prettier-vscode

### LINKED ISSUE #7949: Potential Breaking Change in `resolveConfig`
It seems that 2.0 may have introduced an unintended breaking change in the `resolveConfig` function. This function is used in the Prettier VS Code extension here: https://github.com/prettier/prettier-vscode/blob/master/src/ConfigResolver.ts#L99

Since upgrading to 2.0 we are now seeing errors when trying to resolve configs:

```
Error: Cannot find module 'prettier-config-greenelab' from 'c:\Users\Vincent\Desktop\adage-frontend'
	at Function.Module._resolveFilename (internal/modules/cjs/loader.js:717:15)
	at Function.n.resolve (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:15:584)
	at Object.transform (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\index.js:23283:40)
	at run (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11472:51)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async cacheWrapper (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11300:20)
	at async Explorer.search (c:\Users\Vincent\.vscode\extensions\esbenp.prettier-vscode-4.0.0\node_modules\prettier\third-party.js:11457:22)
	at async Promise.all (index 0)
```
 
For repro steps and more details see: https://github.com/prettier/prettier-vscode/issues/1289

One thing to note, is that everything behaves as expected from the CLI which is odd.

### PR DESCRIPTION
**Pull request by thorn0**
fixes #7949

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/common/resolve.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+let { resolve } = eval("require");
+
+// In the VS Code extension `require` is overridden and `require.resolve` doesn't support the 2nd argument.
+if (resolve.length === 1) {
+  const Module = eval("require")("module");
+  const createRequire = Module.createRequire || Module.createRequireFromPath;
+  resolve = createRequire(__dirname).resolve;
+}
```
Comment (fisker) [inline src/common/resolve.js:10]: I guess unnecessary to do another length check.

Comment (thorn0) [reply src/common/resolve.js:10]: Not sure what you mean.

Comment (fisker) [reply src/common/resolve.js:10]: I mean do we need check `resolve.length` again and throw an error if length still `!=2`, but I guess it's unnecessary.

Comment (thorn0) [reply src/common/resolve.js:10]: Agree.

**Pull request merged**